[2025-08-26T09:00:39.898+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: ml_pipeline.model_training manual__2025-08-26T09:00:21.277583+00:00 [queued]>
[2025-08-26T09:00:39.910+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: ml_pipeline.model_training manual__2025-08-26T09:00:21.277583+00:00 [queued]>
[2025-08-26T09:00:39.910+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 2
[2025-08-26T09:00:39.932+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): model_training> on 2025-08-26 09:00:21.277583+00:00
[2025-08-26T09:00:39.939+0000] {standard_task_runner.py:60} INFO - Started process 226 to run task
[2025-08-26T09:00:39.946+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'ml_pipeline', 'model_training', 'manual__2025-08-26T09:00:21.277583+00:00', '--job-id', '179', '--raw', '--subdir', 'DAGS_FOLDER/ml_pipeline.py', '--cfg-path', '/tmp/tmpvt6whqp6']
[2025-08-26T09:00:39.950+0000] {standard_task_runner.py:88} INFO - Job 179: Subtask model_training
[2025-08-26T09:00:40.049+0000] {task_command.py:423} INFO - Running <TaskInstance: ml_pipeline.model_training manual__2025-08-26T09:00:21.277583+00:00 [running]> on host 1729dc9a5378
[2025-08-26T09:00:40.160+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='sarah' AIRFLOW_CTX_DAG_ID='ml_pipeline' AIRFLOW_CTX_TASK_ID='model_training' AIRFLOW_CTX_EXECUTION_DATE='2025-08-26T09:00:21.277583+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-26T09:00:21.277583+00:00'
[2025-08-26T09:01:21.603+0000] {logging_mixin.py:188} INFO - STDOUT:
[2025-08-26T09:01:21.603+0000] {logging_mixin.py:188} INFO -  Fitting 3 folds for each of 10 candidates, totalling 30 fits
üèÉ View run XGBoost_RandSearch at: https://dagshub.com/SamMebarek/mlops-som.mlflow/#/experiments/0/runs/7fb9a6157866474ab68325e669152ba5
üß™ View experiment at: https://dagshub.com/SamMebarek/mlops-som.mlflow/#/experiments/0
[2025-08-26T09:01:21.604+0000] {logging_mixin.py:188} INFO - STDERR:
[2025-08-26T09:01:21.605+0000] {logging_mixin.py:188} INFO -  2025-08-26 09:00:45,305 - training.utils.common - INFO - YAML charg√© : /opt/***/project/config/config.yaml
2025-08-26 09:00:45,313 - training.utils.common - INFO - YAML charg√© : /opt/***/project/config/params.yaml
2025-08-26 09:00:45,315 - training.utils.common - INFO - R√©pertoire cr√©√© ou existant : /opt/***/project/models
2025-08-26 09:00:45,315 - training.utils.common - INFO - R√©pertoire cr√©√© ou existant : /opt/***/project/models
2025-08-26 09:00:46,079 - __main__ - INFO - Loaded processed data: /opt/***/project/data/processed/clean_data.csv (shape=(108060, 14))
2025-08-26 09:00:46,717 - __main__ - INFO - Split data: train_shape=(86448, 11), test_shape=(21612, 11)
2025/08/26 09:00:46 WARNING mlflow.utils.git_utils: Failed to import Git (the Git executable is probably not on your PATH), so Git SHA is not available. Error: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh(<full-path-to-git-executable>)

All git commands will error until this is rectified.

This initial message can be silenced or aggravated in the future by setting the
$GIT_PYTHON_REFRESH environment variable. Use one of the following values:
    - quiet|q|silence|s|silent|none|n|0: for no message or exception
    - warn|w|warning|log|l|1: for a warning message (logging level CRITICAL, displayed by default)
    - error|e|exception|raise|r|2: for a raised exception

Example:
    export GIT_PYTHON_REFRESH=quiet

2025-08-26 09:00:47,058 - __main__ - INFO - Starting hyperparameter search and training.
2025-08-26 09:01:05,432 - __main__ - INFO - Best params: {'colsample_bytree': 0.9219644132807024, 'gamma': 0.06692515264078093, 'learning_rate': 0.19307895078298942, 'max_depth': 5, 'n_estimators': 86, 'subsample': 0.6271914913054505}
2025-08-26 09:01:05,432 - __main__ - INFO - R^2 score: 0.9998
2025/08/26 09:01:19 INFO mlflow.store.model_registry.abstract_store: Waiting up to 300 seconds for model version to finish creation. Model name: PriceFixerModel, version 76
2025-08-26 09:01:19,695 - __main__ - INFO - Registered model 'PriceFixerModel' from source=mlflow-artifacts:/4829b4e5ce254fea8f678e40a4133671/7fb9a6157866474ab68325e669152ba5/artifacts/xgb_model, run_id=7fb9a6157866474ab68325e669152ba5
2025-08-26 09:01:20,155 - training.repository.repository - INFO - Mod√®le sauvegard√© ‚Üí /opt/***/project/models/xgb_model.pkl
2025-08-26 09:01:20,186 - __main__ - INFO - Local MLflow dir: models/xgb_model_mlflow_manual__2025-08-26T09:00:21.277583+00:00 (size=244004 bytes)
2025-08-26 09:01:20,186 - __main__ - INFO - Model saved locally at: /opt/***/project/models/xgb_model.pkl
2025-08-26 09:01:20,190 - __main__ - INFO - Pushed metrics to Pushgateway (http://pushgateway:9091) status=1 duration_s=34.896 rows=108060 r2=0.9997696380635521 artifact_bytes=484994
[2025-08-26T09:01:21.606+0000] {python.py:201} INFO - Done. Returned value was: None
[2025-08-26T09:01:21.621+0000] {taskinstance.py:1138} INFO - Marking task as SUCCESS. dag_id=ml_pipeline, task_id=model_training, execution_date=20250826T090021, start_date=20250826T090039, end_date=20250826T090121
[2025-08-26T09:01:21.686+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 0
[2025-08-26T09:01:21.732+0000] {taskinstance.py:3280} INFO - 1 downstream tasks scheduled from follow-on schedule check
